services/app-web/src/common-code/proto/healthPlanFormDataProto/toProtoBuffer.ts

```diff
@@ -195,6 +227,11 @@ const toProtoBuffer = (
                                   doc.documentCategories
                               ),
                           })),
+                          rateProgramName: generateRateName(
+                              domainData,
+                              domainData.rateInfos[index],
+                              findRatePrograms(domainData, index)
+                          ),
```

Comment:
Could we pass the `rateInfo` from the map function instead of `domainData.rateInfos[index]` since it would be the same data.

I think we can also just pass all the programs for a given state in the third parameter in `generateRateName()`. `generateRateName()` calls `packageName()` which will use package programs if rate programs do not exists and then `programName()` will find the name of the program from the id.

Reply:
great suggestions; much cleaner now